lib/route.js

```diff
@@ -471,7 +477,7 @@ function buildRouting (options) {
 
     const request = new context.Request(id, params, req, query, childLogger, context)
     const reply = new context.Reply(res, request, childLogger)
-    if (disableRequestLogging === false) {
+    if (resolvedDisableRequestLogging === false) {
```

Comment:
I did not apply same logic for cases like 404, since I assume that error cases should generally be logged, but if consistent behaviour is expected across all of cases, I'm happy to apply this logic everywhere where we log requests